torch_xla/csrc/ops/dynamic_ir.cpp

```diff
@@ -137,6 +137,8 @@ int64_t SizeDiv::getDynamicValue() const {
   const torch::lazy::DimensionNode* dim_node_1 = DimCast(operand(1));
   XLA_CHECK(dim_node_0);
   XLA_CHECK(dim_node_1);
+  XLA_CHECK(dim_node_1->getDynamicValue() != 0)
```

Comment:
nit, save the value of `getDynamicValue` here. I am still debating whether we should cache the value of `GetDynamicValue` here. Each `getDynamicValue` is a recursive call and can be expensive depending on how many layers it has.

Reply:
@JackCaoG according to this [line](https://github.com/pytorch/xla/blob/c2d85a006885a7718affa52feb891e753e811670/torch_xla/csrc/ops/dynamic_ir.cpp#L38-L41), the cache is already in place. Sounds like you have additional compute concern here?

Reply:
I mean the cache for `SizeAddNode`, we currently only have cache for `SizeNode`. `SizeAddNode` can poteianlly cause a regression call but it might be a overkill to cache it here since cpu compute is pretty quick now days.

Reply:
what I mean by save the value of `getDynamicValue` is don't call `dim_node_1->getDynamicValue()` twice here.

Reply:
ideally we should probably have `getDynamicValue()_` that does actual computation and `getDynamicValue()` (in a base class) that calls `getDynamicValue()_`, caches  the value and returns it. wdyt @JackCaoG  ?

Reply:
@Krovatkin It sounds creating a base class with a `getDynamicValue()` would dedup the cache implementation, right? How many such subclasses are there besides SizeAdd/SizeMul/SizeDiv?

Reply:
so you mean something similar to
```
DimensionNode::getDynamicValue_() {
 // everyNode need to implement it
}
DimensionNode::getDynamicValue() {
  if (!cached) {
     cached_val = getDynamicValue_();
  }
  return cached_val?
}
```

yea I guess it works and make things cleaner.

Reply:
SG. Let me add the cache to torch::lazy::DimensionNode then

Reply:
Will do the caching a different PR.

Reply:
> yea I guess it works and make things cleaner.

exactly!!!

Reply:
> How many such subclasses are there besides SizeAdd/SizeMul/SizeDiv?

Potentially as many as methods in SymIntNodeImpl?
20-ish something, I think

Reply:
> @Krovatkin It sounds creating a base class with a getDynamicValue() would dedup the cache implementation, right?

yes, that's correct! it's just a suggestion though, please feel to ignore it